engines/wintermute/ad/ad_block.cpp

```diff
@@ -0,0 +1,55 @@
+/* ResidualVM - A 3D game interpreter
+ *
+ * ResidualVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	 See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+/*
+ * This file is based on WME.
+ * http://dead-code.org/redir.php?target=wme
+ * Copyright (c) 2003-2013 Jan Nedoma and contributors
+ */
+
+#include "engines/wintermute/ad/ad_block.h"
+#include "engines/wintermute/base/base_persistence_manager.h"
+#include "engines/wintermute/base/gfx/opengl/mesh3ds.h"
+
+namespace Wintermute {
+
+//////////////////////////////////////////////////////////////////////////
+AdBlock::AdBlock(BaseGame *inGame) : BaseScriptable(inGame, false, false) {
+	_mesh = nullptr;
+	_active = true;
+	_receiveShadows = false;
+}
+
+//////////////////////////////////////////////////////////////////////////
+AdBlock::~AdBlock() {
+	if (_mesh) {
+		delete _mesh;
+	}
+}
+
+//////////////////////////////////////////////////////////////////////////
+bool AdBlock::persist(BasePersistenceManager *persistMgr) {
+	persistMgr->transferBool(TMEMBER(_active));

```

Comment:
Beware that the extra classes added here will potentially make savegames incompatible between ScummVM and ResidualVM. That's probably something we can get to work in one direction (ScummVM -> ResidualVM, because one is a subset of the other), but not the other. 

Might be worth adding some logic to either fix this where possible (by either using different versioning to actively tag the savegames as ResidualVM-savegames or by not serializing the 3D-classes when not running a 3D-game)

Would you mind at least making a note about looking into the savegame-compatibility issue at some point?

Reply:
Added a comment about in the AdScene::persist, a lot of the 3d stuff is supposed to be persisted there (which is missing at the moment)